src/main/java/uk/ac/ebi/eva/pipeline/configuration/validation/step/VepInputGeneratorStepParametersValidator.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.pipeline.configuration.validation.step;
+
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.JobParametersInvalidException;
+import org.springframework.batch.core.JobParametersValidator;
+import org.springframework.batch.core.job.CompositeJobParametersValidator;
+import org.springframework.batch.core.job.DefaultJobParametersValidator;
+
+import uk.ac.ebi.eva.pipeline.configuration.JobParametersNames;
+import uk.ac.ebi.eva.pipeline.configuration.validation.ConfigRestartabilityAllowValidator;
+import uk.ac.ebi.eva.pipeline.configuration.validation.DbCollectionsVariantsNameValidator;
+import uk.ac.ebi.eva.pipeline.configuration.validation.DbNameValidator;
+import uk.ac.ebi.eva.pipeline.configuration.validation.InputStudyIdValidator;
+import uk.ac.ebi.eva.pipeline.configuration.validation.InputVcfIdValidator;
+import uk.ac.ebi.eva.pipeline.configuration.validation.OutputDirAnnotationValidator;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Validates the job parameters necessary to execute a {@link uk.ac.ebi.eva.pipeline.jobs.steps.VepInputGeneratorStep}
+ * <p>
+ * The parameters OUTPUT_DIR_ANNOTATION, INPUT_STUDY_ID and INPUT_VCF_ID are used to build the VEP input option
+ * {@see uk.ac.ebi.eva.pipeline.configuration.JobOptions#loadPipelineOptions()}
+ */
+public class VepInputGeneratorStepParametersValidator extends DefaultJobParametersValidator {
+
+    public VepInputGeneratorStepParametersValidator() {
+        super(new String[]{JobParametersNames.DB_COLLECTIONS_VARIANTS_NAME, JobParametersNames.DB_NAME,
+                      JobParametersNames.INPUT_STUDY_ID, JobParametersNames.INPUT_VCF_ID,
+                      JobParametersNames.OUTPUT_DIR_ANNOTATION},
+              new String[]{});
+    }
+
+    @Override
+    public void validate(JobParameters parameters) throws JobParametersInvalidException {
+        super.validate(parameters);
+        compositeJobParametersValidator().validate(parameters);
+    }
+
+    private CompositeJobParametersValidator compositeJobParametersValidator() {
+        final List<JobParametersValidator> jobParametersValidators = Arrays.asList(
+                new ConfigRestartabilityAllowValidator(),
```

Comment:
Given that this argument is not mandatory, the associated check should be wrapped inside a conditional. I see at least 2 ways we can do this: 
* Moving it out of the composite validator and adding an "if" condition to the `validate` method. This could be messy for steps with multiple optional arguments.
* Wrapping the `ConfigRestartabilityAllowValidator` with an `OptionalParameterValidator` or something like that, so it can still be declared "inside" the composite validator. This is another indirection level.